_posts/2023-04-04-segment-replication.md

```diff
@@ -0,0 +1,249 @@
+
+You [enable segment replication](https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/segment-replication/configuration/) for your cluster and then per index. Refer to the following diagram of the segment replication process.
+
+<img src="/assets/media/blog-images/2023-04-04-segment-replication/segment-replication.png" alt="Segment replication diagram"/>{: .img-fluid}
+
+## Understanding the trade-offs
+
+With segment replication, you can get the same throughput for ingestion with 9 nodes in a cluster as you would get with 15 nodes with document replication. During testing, our experimental release users reported up to 40% higher throughput with segment replication than with document replication for the same cluster setup.
+
+Compared to document replication, segment replication performs better in OpenSearch cluster deployments with low replica counts, such as those used for log analytics. Segment replication trades CPU for time and networking. The primary is less-frequently, sending larger blocks of data to its replicas. As replica count increases, the primary becomes the bottleneck, doing all of the indexing work, and replicating all of the segments. In our testing, we see an across-the-board improvement for a replica count of 1, decreasing linearly with the replica count. As with all performance results, your mileage will vary! Be sure to test with your own data and queries to determine the benefits for your workload.
+
+For higher replica counts, remote storage integration works better. In remote storage integration, the primary writes segments to an object store, like Amazon Simple Storage Service (S3). Replicas then load the segments from the object store, in parallel, freeing the node with the primary shard from sending out large data blocks to all of the replicas.  We are planning to introduce remote storage integration in a future release.
+
+As with any distributed system, some cluster nodes can fall behind the tolerable or expected throughput levels. Nodes may not be able to catch up to the primary node for various reasons, such as high local search loads or network congestion. To monitor segment replication performance, see [OpenSearch benchmark](https://github.com/opensearch-project/opensearch-benchmark).
+
+## Shard indexing backpressure
+
+When replica nodes start falling behind, the primary node will start applying [shard indexing backpressure](https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/shard-indexing-backpressure/) when ingesting new documents, in an attempt to slow down the indexing. Shard indexing backpressure is a smart rejection mechanism at a per-shard level that dynamically rejects indexing requests when your cluster is under strain. It transfers requests from an overwhelmed node or shard to other nodes or shards that are still healthy.
```

Comment:
Shard indexing backpressure is an existing rejection mechanism.  We have also introduced a new rejection mechanism specific to segrep that is disabled by default. Shard indexing pressure does account level rejections (summing the expected memory consumption per write and rejecting if it hits a threshhold) Segrep backpressure rejects as we recognize replicas have not sync'd to a primary within a certain cutoff time.